client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/launchpad-site-preview.tsx

```diff
@@ -14,25 +22,65 @@ const LaunchpadSitePreview = ( {
 	siteSlug: string | null;
 	flow: string | null;
 } ) => {
+	const site = useSite();
 	const translate = useTranslate();
 	const { globalStylesInUse, shouldLimitGlobalStyles } = usePremiumGlobalStyles();
 
 	const previewUrl = siteSlug ? 'https://' + siteSlug : null;
 	const devicesToShow: Device[] = [ DEVICE_TYPES.COMPUTER, DEVICE_TYPES.PHONE ];
 	let defaultDevice = flow === NEWSLETTER_FLOW ? DEVICE_TYPES.COMPUTER : DEVICE_TYPES.PHONE;
 	const isVideoPressFlow = VIDEOPRESS_FLOW === flow;
+	const isBusinessPlan = site?.plan?.product_slug
+		? isWpComBusinessPlan( site?.plan?.product_slug )
+		: false;
+	const isEcommercePlan = site?.plan?.product_slug
+		? isWpComEcommercePlan( site?.plan?.product_slug )
+		: false;
 
 	if ( isVideoPressFlow ) {
 		const windowWidth = window.innerWidth;
 		defaultDevice = windowWidth >= 1000 ? DEVICE_TYPES.COMPUTER : DEVICE_TYPES.PHONE;
 	}
 
+	const usePreviewSiteLinksQueryEnabled =
+		site?.is_coming_soon && ( isBusinessPlan || isEcommercePlan ) && site?.is_wpcom_atomic;
+
+	const { data: previewLinks, isLoading: isPreviewLinksLoading } = useSitePreviewLinks( {
+		siteId: Number( site?.ID ),
+		isEnabled: usePreviewSiteLinksQueryEnabled ?? false,
+	} );
+
+	const { createLink, isLoading: isCreatingSitePreviewLinks } = useCreateSitePreviewLink( {
+		siteId: Number( site?.ID ),
+	} );
+
+	// Generate preview link for site on business or ecommerce plan
+	// Preview links are only available on these two plans
+	useEffect( () => {
+		if ( previewLinks && Array.isArray( previewLinks ) && previewLinks.length === 0 ) {
+			if ( isBusinessPlan || isEcommercePlan ) {
+				createLink();
+			}
+		}
+	}, [ previewLinks, createLink, isBusinessPlan, isEcommercePlan ] );
+
+	const shareCode = getPreviewCode( previewLinks );
+
+	function getPreviewCode( links: PreviewLink[] | undefined ) {
+		if ( typeof links === 'undefined' ) {
+			return false;
+		}
+		const link = links[ 0 ];
+		return link?.code ?? false;
+	}
```

Comment:
```suggestion
```

We could replace this code by extracting the `shareCode` inside the `formatPreviewUrl`.
```
const shareCode = previewLinks?.[ 0 ]?.code;
```